apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/functions/AllocateAndAddFloatingIpToNode.java

```diff
@@ -275,15 +258,24 @@ public boolean apply(Network input) {
    }
 
 
-   private List<Network> getNetworksFromPoolNames(String regionId, final Optional<Set<String>> poolNames) {
-      return getNetworkApi(regionId).list().concat().filter(new Predicate<Network>() {
-         @Override
-         public boolean apply(Network network) {
-            return poolNames.isPresent() &&
-                   !poolNames.get().isEmpty() &&
-                   poolNames.get().contains(network.getName());
+   /**
+    * Get all suitable networks to allocate a floating ip
+    *
+    * It will prefer networks specified using the poolNames first and then the external networks in the given availability zone
+    */
+   private List<Network> getNetworksFromPoolNamesAndExternal(String regionId, final String availabilityZone, final Optional<Set<String>> poolNames) {
+      if (poolNames.isPresent() && poolNames.get().isEmpty()) return Lists.newArrayList();
```

Comment:
This is not OK. If the pool names are not configured you'll never try to find an external network.
You'll need two different methods, or a method that accepts a filtering predicate.

Reply:
you are right, I will fix it